sql/src/main/java/io/crate/action/sql/TransportSQLAction.java

```diff
@@ -114,7 +115,11 @@ public void pauseProcessed(ResumeHandle resumeable) {
             @Override
             public void finish(RepeatHandle repeatable) {
                 setNextRowResult = RowReceiver.Result.STOP;
-                listener.onResponse(createResponse(analysis, request, rows, startTime));
+                if (analysis.rootRelation() == null) {
```

Comment:
Because it would create an unchecked contract between `expectsAffectedRows` and `rootRelation`. E.g. any time expectsAffectedRows return true it would imply that rootRelation is null. Or the other way around.

And once we add support for `RETURNING` this expectsAffectedRows stuff will become even more confusing.